ibis/expr/types/relations.py

```diff
@@ -2911,29 +2935,20 @@ def join(
         │  106782 │ Leonardo DiCaprio │          5989 │ Leonardo DiCaprio │
         └─────────┴───────────────────┴───────────────┴───────────────────┘
         """
+        from ibis.expr.types.joins import JoinExpr
 
-        _join_classes = {
-            "inner": ops.InnerJoin,
-            "left": ops.LeftJoin,
-            "any_inner": ops.AnyInnerJoin,
-            "any_left": ops.AnyLeftJoin,
-            "outer": ops.OuterJoin,
-            "right": ops.RightJoin,
-            "left_semi": ops.LeftSemiJoin,
-            "semi": ops.LeftSemiJoin,
-            "anti": ops.LeftAntiJoin,
-            "cross": ops.CrossJoin,
-        }
-
-        klass = _join_classes[how.lower()]
-        expr = klass(left, right, predicates).to_expr()
+        # the first participant of the join shouldn't be a self reference
```

Comment:
Ok, but **why**?

Reply:
What I mean here is just to add a slightly more informative comment about why this is the case.

Reply:
Added.